src/main/java/com/rtb/projectmanagementtool/task/TaskData.java

```diff
@@ -42,17 +43,29 @@ public TaskData(Entity entity) {
     projectID = (long) entity.getProperty("projectID");
     name = (String) entity.getProperty("name");
     description = (String) entity.getProperty("description");
-    status = (Status) entity.getProperty("status");
-    users = (HashSet<Long>) entity.getProperty("users");
-    subtasks = (HashSet<Long>) entity.getProperty("subtasks");
+    status = Status.valueOf((String) entity.getProperty("status"));
+    try {
+      users = new HashSet<Long>((ArrayList<Long>) entity.getProperty("users"));
+    } catch (ClassCastException e) {
+      users = (HashSet<Long>) entity.getProperty("users");
```

Comment:
I'm curious about which scenarios where the first one would fail, but the second would be ok?  IIUC the datastore schema would hold the same data type (unless if you change), which I think would mean there wouldn't be a scenario where the same Property would be 2 different types (an ArrayList and a HashSet)?

Reply:
It's because while testing, I would create an entity from scratch using HashSets and transform it to a TaskData object without putting it on datastore. Therefore, it wouldn't become an ArrayList and would remain a HashSet. Would it be better to change the test so that an entity was created with an ArrayList? Or, should I put the entity in datastore first so that datastore would transform it for me? Or, should I leave it the way it is, with the TaskData class taking care of either option?

Reply:
I see what you mean.  Generally you'd want your test to behave as close to prod behavior as possible.  If prod has entities with ArrayLists on them, you should also put together your tests to work with ArrayLists.  This will allow you to exercise your list->set conversion.  It'll also prevent you from having test-only code in your prod code.

Alternative approach would be to switch to ArrayList all together. This will make your Data more closely mimic what the actual DataStore data looks like.  Although you do lose the convenience of quickly looking up individual values from the set.  If you see yourself having to lookup specific values from the list/set, it could be worth it to do this extra conversion to Set.  If most of the time though you'll be iterating the whole list, maybe keep it as a List then only convert to Set when needed.

Reply:
I think that the majority of actions performed on users and subtasks would be iterating, adding, and removing. In that case, an ArrayList would probably be easier to work with overall.

Reply:
I'm curious, can you elaborate a bit more on your thoughts regarding set vs. list?  (convenience, performance, efficiency, storage, etc..)